### PR TITLE
[WIP] Adding util to get Numpy 3D array of RGBA values

### PR DESCRIPTION
**Pull request by ShreyasFadnavis**
The name is explanatory

### CODE REVIEW

File: fury/utils.py [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 from __future__ import division, print_function, absolute_import
 
+import os
 import sys
 import numpy as np
 import vtk
 from vtk.util import numpy_support
 from scipy.ndimage import map_coordinates
 from fury.colormap import line_colors
+from nibabel.tmpdirs import InTemporaryDirectory
```
Comment (skoudoro) [inline fury/utils.py]: `from fury.tmpdirs import InTemporaryDirectory` instead

File: fury/utils.py [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+from dipy.utils.optpkg import optional_package
```
Comment (skoudoro) [inline fury/utils.py]: `from fury.optpkg import optional_package` instead

File: fury/utils.py [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+_, have_imread, _ = optional_package('Image')
```
Comment (skoudoro) [inline fury/utils.py]: we do not need this. you can remove it

File: fury/utils.py [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
+matplotlib, have_mpl, _ = optional_package("matplotlib")
+
+if have_imread:
+    from scipy.misc import imread
```
Comment (skoudoro) [inline fury/utils.py]: we do not need this too.  Instead, `from fury.io import load_image`

File: fury/utils.py [hunk 5]
```diff
@@ -702,3 +710,43 @@ def rotate(actor, rotation=(90, 1, 0, 0)):
         prop3D.SetPosition(newTransform.GetPosition())
         prop3D.SetScale(newTransform.GetScale())
         prop3D.SetOrientation(newTransform.GetOrientation())
+
+
+def matplotlib_figure_to_numpy(fig, dpi=100, fname=None, flip_up_down=True,
+                               transparent=False):
+    r""" Convert a Matplotlib figure to a 3D numpy array with RGBA channels
+    Parameters
```
Comment (skoudoro) [inline fury/utils.py]: need empty line between the description and parameters

File: fury/utils.py [hunk 6]
```diff
@@ -702,3 +710,43 @@ def rotate(actor, rotation=(90, 1, 0, 0)):
...
+    ----------
+    fig : obj,
```
Comment (skoudoro) [inline fury/utils.py]: can you remove the coma?

File: fury/utils.py [hunk 7]
```diff
@@ -702,3 +710,43 @@ def rotate(actor, rotation=(90, 1, 0, 0)):
...
+        Dots per inch
+    fname : str
+        If ``fname`` is given then the array will be saved as a png to this
+        position.
+    flip_up_down : bool
+        The origin is different from matlplotlib default and VTK's default
+        behaviour (default True).
+    transparent : bool
+        Make background transparent (default False).
+    Returns
```
Comment (skoudoro) [inline fury/utils.py]: need empty line before `Returns`

File: fury/tests/test_utils.py [hunk 8]
```diff
@@ -5,9 +5,12 @@
                         vtk_matrix_to_numpy,
                         numpy_to_vtk_matrix,
                         get_grid_cells_position,
-                        rotate, vtk)
+                        rotate, vtk, matplotlib_figure_to_numpy)
 from fury import actor, window, utils
 from fury.decorators import xvfb_it
+from dipy.utils.optpkg import optional_package
```
Comment (skoudoro) [inline fury/tests/test_utils.py]: See comment below

File: fury/tests/test_utils.py [hunk 9]
```diff
@@ -5,9 +5,12 @@
...
+_, have_imread, _ = optional_package('Image')
```
Comment (skoudoro) [inline fury/tests/test_utils.py]: see comment below

File: fury/tests/test_utils.py [hunk 10]
```diff
@@ -256,5 +259,14 @@ def test_rotate(interactive=False):
         npt.assert_equal(red_sum_new > red_sum, True)
 
 
+def test_matplotlib_figure_to_numpy():
```
Comment (skoudoro) [inline fury/tests/test_utils.py]: add a `skipif(not have_mpl)` ,  you can look at `test_actor.py`

File: fury/tests/test_utils.py [hunk 11]
```diff
@@ -256,5 +259,14 @@ def test_rotate(interactive=False):
...
+    fig = matplotlib.pyplot.figure()
+    matplotlib.pyplot.plot([1, 2, 3])
+    matplotlib.pyplot.close()
+    arr = matplotlib_figure_to_numpy(fig)
+    test_arr = np.zeros_like(arr)
+    npt.assert_raises(AssertionError, npt.assert_array_equal, arr, test_arr)
```
Comment (skoudoro) [inline fury/tests/test_utils.py]: I do not understand this. Can you explain me?

Comment (ShreyasFadnavis) [reply fury/tests/test_utils.py]: Do the comments help? Is there some better way to test this? Newbie at testing viz.. Any advice?

**Comment by skoudoro** [comment]
closing in favor or #124

**Pull request closed without merging**